clients/src/test/java/org/apache/kafka/common/security/authenticator/SaslAuthenticatorTest.java

```diff
@@ -2128,7 +2116,7 @@ private void createAndCheckClientAuthenticationFailure(SecurityProtocol security
             String mechanism, String expectedErrorMessage) throws Exception {
         ChannelState finalState = createAndCheckClientConnectionFailure(securityProtocol, node);
         Exception exception = finalState.exception();
-        assertTrue("Invalid exception class " + exception.getClass(), exception instanceof SaslAuthenticationException);
+        assertTrue(exception instanceof SaslAuthenticationException, "Invalid exception class " + exception.getClass());
```

Comment:
Personally, checking the class type is more readable to me. Also, it can generate meaningful error message (the custom message can be removed). If this comment is valid, we can address it by another PR (as there are a lot of occurances)

Reply:
Yeah, for existing code improvements (versus code introduced by this change), let's do it via a different PR. For this particular issue, we can probably wait for JUnit 5.8 and use:

> * New assertInstanceOf methods as a replacement for assertTrue(obj instanceof X) which provide better error messages comparable to those of assertThrows.

Reply:
Nice suggestion. Open a ticket (https://issues.apache.org/jira/browse/KAFKA-12187) for it.